src/coreclr/jit/emitxarch.cpp

```diff
@@ -4272,14 +4284,24 @@ void emitter::emitIns_Mov(instruction ins, emitAttr attr, regNumber dstReg, regN
     assert(size <= EA_32BYTE);
     noway_assert(emitVerifyEncodable(ins, size, dstReg, srcReg));
 
+    UNATIVE_OFFSET sz = emitInsSizeRR(ins, dstReg, srcReg, attr);
+
     if (canSkip && (dstReg == srcReg))
     {
         switch (ins)
         {
             case INS_mov:
             {
-                // These instructions have no side effect and can be skipped
-                return;
+                // These instructions might have a side effect in the form of a
+                // byref liveness update, so preserve them but emit nothing
+
+                if (!EA_IS_BYREF(attr))
+                {
+                    return;
+                }
+
+                sz = 0;
```

Comment:
I wonder if, in this case, we should change the `ins` to some pseudo-instruction `INS_mov_elided`, to make sure it is distinguished from normal `mov`. That might make some other checks cleaner, as well.

Reply:
I'll see about adding a pseudo-instruction.